### ISSUE TITLE
[java] NullAssignment should not report assigning `null` to a final field in a constructor

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#4608]
**Affects PMD Version:** 
6.55.0

**Rule:**
[NullAssignment](https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_java_errorprone.html#nullassignment)

**Description:**
If you have a `final` field that might be `null` or not, depending on the called constructor or depending on the arguments given to the constructor, you **must** set it to `null` explicitly in the constructor.
If you set it at declaration time, you cannot set it to a different value.
So assigning `null` to a `final` field in a constructor should not trigger a finding.

**Code Sample demonstrating the issue:**

```java
public class RestrictionChainElement {
    private final Class<? extends Restriction<?>> restriction;

    RestrictionChainElement() {
        this.restriction = null;
    }

    public RestrictionChainElement(Class<? extends Restriction<?>> restriction) {
        this.restriction = requireNonNull(restriction);
    }
}
```

**Expected outcome:**
PMD reports a violation at line 5, but that's wrong. That's a false positive.

**Running PMD through:** 
Gradle

**Comment by jsotuyod** [comment]
PMD 7 already handles this properly.

Tests included:
https://github.com/pmd/pmd/blob/027fb4b9f435b00031dbc5de867ae4e3f75c3ccc/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/NullAssignment.xml#L236-L251

https://github.com/pmd/pmd/blob/027fb4b9f435b00031dbc5de867ae4e3f75c3ccc/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/NullAssignment.xml#L133-L144
